src/main/java/org/codehaus/groovy/runtime/MetaClassHelper.java

```diff
@@ -1016,4 +1018,21 @@ public static String convertPropertyName(String prop) {
         }
         return BeanUtils.decapitalize(prop);
     }
+
+    // GROOVY-9075
+    public static void checkMetaClassProperty(MetaClass oldValue, Object newValue) {
+        if (null == newValue) return;
+
+        if (!(newValue instanceof MetaClass)) {
+            throw new IllegalArgumentException("the new metaClass[" + newValue.getClass() + "] is not an instance of " + MetaClass.class);
```

Comment:
I think the common convention is upper case for first letter.

Reply:
OK. I will refine the message later.

Reply:
OK. I will refine the message later.